src/match_identifiers.cpp

```diff
@@ -96,50 +112,69 @@ void match_identifiers(
 	// scan body
 	assert(ast->m_body->type() == ast_type::Block);
 	auto body = static_cast<TypedAST::Block*>(ast->m_body.get());
-	for (auto& child : body->m_body)
-		match_identifiers(child.get(), env);
+	for (auto& child : body->m_body) {
+		CHECK_AND_RETURN(match_identifiers(child.get(), env));
+	}
 
```

Comment:
Nit: the for loop braces here are not consistent with other for loops in the diff